src/scripts/pokemons/PokemonList.ts

```diff
@@ -16437,6 +16439,7 @@ pokemonList.forEach(p => {
         p.evolutions?.forEach(evo => pokemonDevolutionMap[evo.getEvolvedPokemon()] = evo.basePokemon);
     }
     p.nativeRegion = p.nativeRegion || GameConstants.TotalPokemonsPerRegion.findIndex(maxRegionID => maxRegionID >= p.id);
+    pokemonNameIndex[p.name.toLowerCase()] = p;
```

Comment:
Would it be more memory efficient if we just linked to the index of the Pokemon in the array?:
```js
pokemonList.forEach((p, index) => {
    ...
    pokemonNameIndex[p.name.toLowerCase()] = index;
}

const pokemonMap: any = new Proxy(pokemonList, {
    ...
    return pokemon[pokemonNameIndex[prop.toLowerCase()]] || pokemon[prop] || pokemon.find(p => p.id == 0);
```

Reply:
I don't think so, javascript won't actually store the object in pokemonIndex, just a pointer to it. If `p` was some large string then using the index would be more memory efficient, but for objects I don't think it will make any difference.

You are welcome to change it to using the pokemonList index for peace of mind though, I only did it this way because it was slightly less characters to write :sweat_smile: